lib/hook/consumers/github.ex

```diff
@@ -0,0 +1,140 @@
+defmodule Accent.Hook.Consumers.GitHub do
+  @moduledoc """
+  From a project’s integration association and a webhook event from GitHub,
+  sync and add translation as if the operations were made manually by a user.
+
+  """
+  use Accent.Hook.EventConsumer, subscribe_to: [Accent.Hook.Producers.GitHub]
+
+  alias Accent.{Document, Repo, Version}
+  alias Accent.Hook.Consumers.GitHub.AddTranslations
+  alias Accent.Hook.Consumers.GitHub.Sync
+  alias Accent.Hook.Context
+  alias Accent.Plugs.MovementContextParser
+  alias Accent.Scopes.Document, as: DocumentScope
+  alias Accent.Scopes.Version, as: VersionScope
+
+  def handle_events(events, _from, state) do
+    Enum.each(events, &handle_event/1)
+
+    {:noreply, [], state}
+  end
+
+  defp handle_event(%Context{user: user, project: project, event: "push", payload: payload}) do
+    payload[:ref]
+    |> ref_to_version(payload[:default_ref], project)
+    |> sync_and_add_translations(project, user, payload)
+  end
+
+  defp sync_and_add_translations({:ok, version}, project, user, payload) do
+    repo = payload[:repository]
+    token = payload[:token]
+
+    ref = (version && version.tag) || payload[:default_ref]
+    configs = fetch_config(repo, token, ref)
+    trees = fetch_trees(repo, token, ref)
+
+    revisions =
+      project
+      |> Ecto.assoc(:target_revisions)
+      |> Repo.all()
+      |> Repo.preload(:language)
+
+    Sync.persist(trees, configs, project, user, payload, version)
+
+    Enum.each(revisions, fn revision ->
+      AddTranslations.persist(trees, configs, project, user, revision, payload, version)
+    end)
+  end
+
+  defp sync_and_add_translations(_, _, _, _), do: :ok
```

Comment:
Since this impl cover the fact that version wasn't parsed correctly either from default or version_from_ref, shall we add some log for this particular path?  Or maybe there is a mechanism in place to surface warning/errors to user in the dashboard?

Reply:
It’s ok to do nothing if the version is not the default or a matching version. When we receive a ref like `feature/my-new-feature` it’s the expected behaviour  to ignore the push.

Reply:
Ah! Good point.